server/src/main/java/org/apache/druid/client/CachingClusteredClient.java

```diff
@@ -126,11 +128,13 @@ public class CachingClusteredClient implements QuerySegmentWalker
   private final ForkJoinPool pool;
   private final QueryScheduler scheduler;
   private final ServiceEmitter emitter;
+  private final BrokerViewOfCoordinatorConfig brokerViewOfCoordinatorConfig;
 
   @Inject
   public CachingClusteredClient(
       QueryRunnerFactoryConglomerate conglomerate,
       TimelineServerView serverView,
+      BrokerViewOfCoordinatorConfig brokerViewOfCoordinatorConfig,
```

Comment:
why does `CachingClusteredClient` need this to just pass into `ServerSelector` instead of creating `ServerSelector` with it? And since we pass the query into `pick` anyway it seems like nothing here would have to change if we did that instead

Reply:
Thanks for this suggestion, I missed this earlier as `BrokerViewOfCoordinatorConfig` was being used to create the filters. This cleaned up a lot of the code.